api/helpers/gamestate/moves/resolve-four/validate.js

```diff
@@ -0,0 +1,64 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Validate request to resolve a four',
+
+  description: 'Verifies whether a request to resolve a four is legal, throwing explanatory error if not.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      descriptions: 'Object containing the current game state',
+      required: true,
+    },
+    /**
+     * @param { Object } requestedMove - Object describing the request to resolve a four
+     * @param { String } requestedMove.cardId1 - First card to be discarded
+     * @param { String } requestedMove.cardId2 - Second card to be discarded
+     * @param { MoveType.RESOLVE_FOUR } requestedMove.moveType - Specifies that this is a Resolve Four move
+     */
+    requestedMove: {
+      type: 'ref',
+      description: 'Object containing data needed for current move',
+      required: true,
+    },
+    playedBy: {
+      type: 'number',
+      description: 'Player number of player requesting move',
+      required: true,
+    },
+  },
+  sync: true,
+  fn: ({ requestedMove, currentState, playedBy }, exits) => {
+    try {
+      if (currentState.turn % 2 === playedBy) {
+        throw new Error('game.snackbar.oneOffs.four.notYourTurn');
+      }
+
+      if (currentState.phase !== GamePhase.RESOLVING_FOUR) {
+        throw new Error('game.snackbar.oneOffs.four.notResolvingFourPhase');
+      }
+
+      const playerKey = playedBy ? 'p1' : 'p0';
+      const { cardId1, cardId2 } = requestedMove;
+      let selectedCard1, selectedCard2;
+
+      currentState[playerKey].hand.forEach(card =>{
```

Comment:
the playerKey variable is actually less elegant than just assigning to player directly imo. That's also more consistent with the other moves. I'd like to update the `points/validate` move to reflect this as well for consistency.

```suggestion
      const player = playedBy ? result.p1 : result.p0;
      const { cardId1, cardId2 } = requestedMove;
      let selectedCard1, selectedCard2;

      player.hand.forEach(card =>{
```

That said as @Haviles04 said it may be sensible to use two `.find()` calls instead of the `forEach` just for clarity. It's an extra loop, but a fairly clear line.

I'm not opposed to the single-loop approach here with the block-soped `selectedCard1` and `selectedCard2` variables, either. If we keep the single-loop like you have here, let's reformat the conditionals so the blocks live on their own lines. Single line conditionals like that are too cluttered imo.